src/virt.c

```diff
@@ -2687,12 +2768,26 @@ static void free_domains(struct lv_read_state *state) {
 
 static int add_domain(struct lv_read_state *state, virDomainPtr dom,
                       bool active) {
-
   int new_size = sizeof(state->domains[0]) * (state->nr_domains + 1);
+  const char *domname = virDomainGetName(dom);
```

Comment:
IMHO, that is not the best solution to put ignorelist check into `add_domain`.
It is not intuitive that `add_domain` is smart and can decide to skip adding.

I think that will be better to move this check into another function and call it before `add_domain`.

Can you please update plugin code?

Many thanks!

Reply:
it was tempting to add logic here because all needed fields were already fetched in add_domain but it is no problem to get domain name separately and skip domain before 'add_domain' call. I'll refactor this